src/snowcli/cli/nativeapp/commands.py

```diff
@@ -72,3 +69,21 @@ def app_bundle(
     manager = NativeAppManager(project_path)
     manager.build_bundle()
     return MessageResult(f"Bundle generated at {manager.deploy_root}")
+
+
+@app.command("run")
+@with_output
+@global_options
```

Comment:
With the current implementation of this command, it only accepts `@global_options` and not `@global_options_with_connection`. But as I recall from our recent meeting, we want to support `@global_options_with_connection`, and provide priority to `-r role` if specified, over the role in `snowflake.yml`. Is that correct?

Reply:
Yes, `-r role` will override role from `config.toml`

Reply:
So if we need to alter connection configurtion we should use `global_options_with_connection`

Reply:
From our team's conversation - for `run` we should not have `global_options_with_connection` especially `-role`. This is because `run` uses two different roles - one for app pkg and one for app, and we would not have clear distinction on which role to actually override, and it would also cause confusion for the user. 
Once we start introducing atomic commands, then we can include the scope there.

Reply:
I think without `global_options_with_connection`, the default connection isn't set up! At least I couldn't get `snow app run` to work without changing to that. What's been your experience?

Reply:
Shouldn't the default connection be coming from `config.toml` ? `global_options_with_connection` is about overrides to an existing default, afaik.